lib/team_statistics.rb

```diff
@@ -76,53 +77,59 @@ def fewest_goals_scored(team_id)
     end.goals
   end
 
-  def get_all_team_games(team_id)
-    games.find_all do |game|
-      game.home_team_id == team_id.to_i || game.away_team_id == team_id.to_i
-    end
+  def favorite_opponent(team_id)
+    rival_id = opponent_win_count(team_id).sort_by { |k, v| v[:average] }.first[0]
+    rival_team = find_team(rival_id.to_s)
+    rival_team.teamname
   end
 
-  def get_games_by_id(team_id)
-    get_all_team_games(team_id).map do |game|
-      game.game_id
+  def rival(team_id)
```

Comment:
This looks good, we reviewed on a call in person, and this def runs much faster than the previous way.